src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs

```diff
@@ -24,22 +30,55 @@ public ModalImporterPresenter(IModalImporterView view, IDataSetToDataRepositoryM
          _dataRepositoryMapper = dataRepositoryMapper;
       }
 
-      public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(IImporterPresenter presenter, IReadOnlyList<MetaDataCategory> metaDataCategories, IReadOnlyList<ColumnInfo> columnInfos, DataImporterSettings dataImporterSettings)
+      public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(
+         IImporterPresenter presenter, 
+         IReadOnlyList<MetaDataCategory> metaDataCategories, 
+         IReadOnlyList<ColumnInfo> columnInfos, 
+         DataImporterSettings dataImporterSettings
+      )
       {
          List<DataRepository> result = new List<DataRepository>();
          _view.FillImporterPanel(presenter.BaseView);
          var id = Guid.NewGuid().ToString();
          presenter.OnTriggerImport += (s, d) =>
          {
             var i = 0;
-            foreach (var pair in d.DataSource.DataSets.KeyValues)
+            foreach (var dataSet in d.DataSource.DataSets.SelectMany(ds => ds.Data))
             {
-               foreach (var data in pair.Value.Data)
+               var dataRepo = _dataRepositoryMapper.ConvertImportDataSet(d.DataSource.DataSetAt(i++));
+               dataRepo.ConfigurationId = id;
+               var molecule = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation);
+               var moleculeDescription = (metaDataCategories?.FirstOrDefault(md => md.Name == dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)?.ListOfValues.FirstOrDefault(v => v.Key == dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMoleculeInformation)))?.Value;
+               var molecularWeightDescription = dataRepo.ExtendedPropertyValueFor(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation);
+
+
+               if (moleculeDescription != null)
+               {
+                  if (string.IsNullOrEmpty(molecularWeightDescription))
+                  {
+                     molecularWeightDescription = moleculeDescription;
+                     if (!string.IsNullOrEmpty(dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation))
+                     dataRepo.ExtendedProperties.Add(new ExtendedProperty<string>() { Name = dataImporterSettings.NameOfMetaDataHoldingMolecularWeightInformation, Value = moleculeDescription });
+                  }
+                  else
+                  {
+                     double molWeight;
+                     double moleculeMolWeight;
+                     double.TryParse(moleculeDescription, out moleculeMolWeight);
+                     double.TryParse(molecularWeightDescription, out molWeight);
+                     if (Math.Abs(moleculeMolWeight - molWeight) > Constants.DOUBLE_RELATIVE_EPSILON)
+                        throw new InconsistenMoleculeAndMoleWeightException();
```

Comment:
It's actually an _absolute_ error check, not _relative_. 
Does not really matter here because MW is >1 anyway (thus if the relative error is above tolerance => absolute error will be above tolerance as well)

@msevestre Don't we have a method `EqualWithinRelativeTolerance `somewhere? Or was it only in the test framework?

Reply:
@Yuri05 maybe this:
OSPSuite.Core.Domain.Services.ValueComparer.AreValuesEqual

I am committing the change next.

Reply:
>OSPSuite.Core.Domain.Services.ValueComparer.AreValuesEqual

Yep, this one :)

Reply:
> > OSPSuite.Core.Domain.Services.ValueComparer.AreValuesEqual
> 
> Yep, this one :)

Good, I have just committed the fix. Thanks for the tip.

Reply:
almost there. should be `if NOT AreValuesEqual` :)

Reply:
> almost there. should be `if NOT AreValuesEqual` :)

Sure... I updated with the solution now... I hope ;)